suite/tests/CMakeLists.txt

```diff
@@ -4100,10 +4104,10 @@ if (BUILD_CLIENTS)
       torunonly_drcacheoff(fork linux.fork "" "" "")
     endif ()
 
-    # Test reading a legacy pre-interleaved file.
+    # Test reading a legacy pre-interleaved file in thread-sharded mode.
```

Comment:
So now the user needs to remember to specify -no_core_sharded/-no_core_serial when they really want thread-sharded? But not if some tool doesn't prefer core-sharded?

I don't see how this is a strict improvement over the current state of things.

Reply:
The current state is broken IMHO.  Do you have an alternative suggestion for how to solve it?

Reply:
Offline discussion settled on the preferred mode scheme of this PR plus failing on a mix.